### PR TITLE
Ip feature

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -75,46 +156,82 @@ func (s *Scheduler) ReserveResources(hostID string, force bool, resourceRequests
...
+			pool := p.(*PortResourcePool)
+			request := rr.(PortBindingResourceRequest)
+			result, er := PortReserve(pool, request)
+			logrus.Infof("Host-UUID %v, PortPool Map %v", hostID, pool.PortBindingMap)
+			if er != nil {
+				err = er
+				portsRollback = result
+				break L
+			} else {
+				data[request.Resource] = result
```
Comment (cjellick) [inline scheduler/scheduler.go]: BUG! If I send more than one port request, I only get 1 back to program in cattle

Comment (cjellick) [reply scheduler/scheduler.go]: Logs proving:
```
time="2017-01-24T21:55:55Z" level=info msg="Received event: Name: scheduler.reserve, Event Id: 8eebf538-bec1-45d9-b1d2-fd1a0b9b937e, ResourceId : 1i54" 

time="2017-01-24T21:55:55Z" level=info msg="Reserving [{Resource:portReservation InstanceID:54 PortRequests:[{PrivatePort:9090 IPAddress: PublicPort:56523}]} {Resource:instanceReservation Amount:1} {Resource:portReservation InstanceID:55 PortRequests:[{PrivatePort:9091 IPAddress: PublicPort:53696}]} {Resource:instanceReservation Amount:1}] for c7886390-58f7-4048-bb7a-c226027b0a01. Force=false" 

time="2017-01-24T21:55:55Z" level=info msg="Public port 56523 reserved for ip address 0.0.0.0" 

# Reserving first port
time="2017-01-24T21:55:55Z" level=info msg="Host-UUID c7886390-58f7-4048-bb7a-c226027b0a01, PortPool Map map[0.0.0.0:map[500:true 4500:true 56523:true]]" 

# Reserving second port
time="2017-01-24T21:55:55Z" level=info msg="Public port 53696 reserved for ip address 0.0.0.0" 

time="2017-01-24T21:55:55Z" level=info msg="Host-UUID c7886390-58f7-4048-bb7a-c226027b0a01, PortPool Map map[0.0.0.0:map[4500:true 56523:true 53696:true 500:true]]" 

# See here we only have one port in the response?
time="2017-01-24T21:55:55Z" level=info msg="Reply: Name: reply.4683323719371391497, PreviousIds: [8eebf538-bec1-45d9-b1d2-fd1a0b9b937e], ResourceId: 1i54, Data: map[portReservation:map[instanceID:55 allocatedIPs:[map[publicPort:53696 privatePort:9091 allocatedIP:0.0.0.0]]]]."
```

Comment (StrongMonkey) [reply scheduler/scheduler.go]: bug fix fixed the bug you mentioned

**Comment by StrongMonkey** [comment]
this latest commit will make that test passed along with your cattle PR. @cjellick

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -57,32 +57,49 @@ func (p *PortResourcePool) IsIPQualifiedForRequests(ip string, specs []PortSpec)
...
+				logrus.Infof("Port %v is reserved for IP %v for ghost map", port, ip)
+				return nil
 			}
-			if success {
+			if !p.GhostMap[ip][port] {
+				p.GhostMap[ip][port] = true
+				logrus.Infof("Port %v is reserved for IP %v for ghost map", port, ip)
 				return nil
 			}
+			return errors.Errorf("Port %v is already used in ip %v in ghost map", port, ip)
```
Comment (cjellick) [inline scheduler/scheduler.go]: This error message might make its way back to the user, so we cannot use the term "ghost map". Just drop the phrase "in ghost map" from the end.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**